From 30f86d763debf8ce2b4a5171f987f1b23abec604 Mon Sep 17 00:00:00 2001
From: stone.liu <stone.liu@dniserver.dnish.net>
Date: Fri, 28 Oct 2011 07:31:04 -0400
Subject: [PATCH 104/137] nf_conntrack_proto_tcp: Add if(refresh) before nf_ct_refresh_acct

Since refresh is 1, and no one will change it.
Add this if conditino would still run like original Linux Kernel.

DNI Common Module Patch Review Checklist
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Here are some basic things that I (Jonathan Chang) have checked
in order to ensure the quality of common modules.

1: The syntax and comment are compliant.

2: No memory leak.
   * memory allocated by malloc() will be freed
   * memory allocated by calloc() will be freed

3: No deadlock such as
   * No function call deadlock.
   * No semaphore deadlock.
   * No tag or label deadlock.

4: No stack overflow caused by
   * infinite recursion
   * too large local array variables

5: coding completeness, such as
   * conditions completeness
   * functions completeness
   * update completeness
   * memory resources completeness
   * sessions resources completeness
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Jack Lin <jack.lin@dniserver.dnish.net>
Signed-off-by: Jonathan Chang <jonathan@dniserver.dnish.net>

Conflicts:

	net/netfilter/nf_conntrack_proto_tcp.c

Signed-off-by: stone.liu <stone.liu@dniserver.dnish.net>
---
 net/netfilter/nf_conntrack_proto_tcp.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/net/netfilter/nf_conntrack_proto_tcp.c b/net/netfilter/nf_conntrack_proto_tcp.c
index 1dde473..d201411 100644
--- a/net/netfilter/nf_conntrack_proto_tcp.c
+++ b/net/netfilter/nf_conntrack_proto_tcp.c
@@ -1077,6 +1077,7 @@ static int tcp_packet(struct nf_conn *ct,
 #if defined(CONFIG_RTL_NF_CONNTRACK_GARBAGE_NEW)
 	nf_ct_refresh_acct_tcp(ct, ctinfo, skb, timeout, old_state, new_state);
 #else
+	if(refresh)
 	nf_ct_refresh_acct(ct, ctinfo, skb, timeout);
 #endif
 	//nf_ct_refresh_acct(ct, ctinfo, skb, timeout);
-- 
1.7.4.1

